### PR TITLE
Add `RefCallback`

### PR DESCRIPTION
**Pull request by aerialist7**
cc @turansky @sgrishchenko

### CODE REVIEW

File: kotlin-react-dom/src/main/kotlin/react/dom/RDOMBuilder.kt [hunk 1]
```diff
@@ -55,15 +55,15 @@ interface RDOMBuilder<out T : Tag> : RBuilder {
...
+    var ref: Ref<*>
         @Deprecated(message = "Write-only property", level = DeprecationLevel.HIDDEN)
         get() = error("")
         set(value) {
             domProps.ref = value
         }
 
-    fun ref(handler: (dynamic) -> Unit) {
-        domProps.ref(handler)
+    fun ref(handler: RefCallback<*>) {
```
Comment (turansky) [inline kotlin-react-dom/src/main/kotlin/react/dom/RDOMBuilder.kt:65]: Redundant?

File: kotlin-react/src/main/kotlin/react/RBuilder.kt [hunk 2]
```diff
@@ -237,15 +237,15 @@ interface RElementBuilder<out P : RProps> : RBuilder {
...
+    var ref: Ref<*>
         @Deprecated(message = "Write-only property", level = DeprecationLevel.HIDDEN)
         get() = error("")
         set(value) {
             attrs.ref = value
         }
 
-    fun ref(handler: (dynamic) -> Unit) {
-        attrs.ref(handler)
+    fun ref(handler: RefCallback<*>) {
```
Comment (turansky) [inline kotlin-react/src/main/kotlin/react/RBuilder.kt:247]: Redundant?

**Review by turansky — CHANGES_REQUESTED** [review]

File: kotlin-react/src/main/kotlin/react/RBuilder.kt (hunk 2 above)
Comment (aerialist7) [reply kotlin-react/src/main/kotlin/react/RBuilder.kt]: Fixed

File: kotlin-react-dom/src/main/kotlin/react/dom/RDOMBuilder.kt (hunk 1 above)
Comment (aerialist7) [reply kotlin-react-dom/src/main/kotlin/react/dom/RDOMBuilder.kt]: Fixed

**Review by turansky — APPROVED** [review]

**Comment by Leonya** [comment]
Released in pre.220

**Pull request merged**